worlds/lufia2ac/Options.py

```diff
@@ -0,0 +1,537 @@
+    range_start = 0
+    range_end = 9980
+    default = 9980
+    special_range_cutoff = 1
+    special_range_names = {
+        "default": 9980,
+        "scale": 0,
+    }
+
+    @staticmethod
+    def scale(final_floor: int) -> int:
+        return final_floor * 100 + 80
+
+
+class PartyStartingLevel(LevelMixin, Range):
+    """The starting level of your party members.
+
+    Supported values: 1 – 99
+    Default value: 1 (same as in an unmodified game)
+    """
+
+    display_name = "Party starting level"
+    range_start = 1
+    range_end = 99
+    default = 1
+
+    @property
+    def xp(self) -> int:
+        return self._to_xp(self.value, capsule=False)
+
+
+class RunSpeed(Choice):
+    """Modifies the game to allow you to move faster than normal when pressing the Y button.
+
+    Supported values: disabled, double, triple, quadruple
+    Default value: disabled (same as in an unmodified game)
+    """
+
+    display_name = "Run speed"
+    option_disabled = 0x08
+    option_double = 0x10
+    option_triple = 0x16
+    option_quadruple = 0x20
+    default = option_disabled
+
+
+class ShuffleCapsuleMonsters(Toggle):
+    """Shuffle the capsule monsters into the multiworld.
+
+    Supported values:
+    false — all 7 capsule monsters are available in the menu and can be selected right away
+    true — you start without capsule monster; 7 new "items" are added to your pool and shuffled into the multiworld;
+        when one of these items is found, the corresponding capsule monster is unlocked for you to use
+    Default value: false (same as in an unmodified game)
+    """
+
+    display_name = "Shuffle capsule monsters"
+
+    @property
+    def unlock(self) -> int:
+        return 0b00000000 if self.value else 0b01111111
+
+
+class ShufflePartyMembers(Toggle):
+    """Shuffle the party members into the multiworld.
+
+    Supported values:
+    false — all 6 optional party members are present in the cafe and can be recruited right away
+    true — only Maxim is available from the start; 6 new "items" are added to your pool and shuffled into the
+        multiworld; when one of these items is found, the corresponding party member is unlocked for you to use
+    Default value: false (same as in an unmodified game)
+    """
+
+    display_name = "Shuffle party members"
+
+    @property
+    def unlock(self) -> int:
+        return 0b00000000 if self.value else 0b11111100
+
+
+class StartingCapsule(Choice):
+    """The capsule monster you start the game with.
+
+    Only has an effect if shuffle_capsule_monsters is set to false.
+    Supported values: jelze, flash, gusto, zeppy, darbi, sully, blaze
+    Default value: jelze
+    """
+
+    display_name = "Starting capsule monster"
+    option_jelze = 0x00
+    option_flash = 0x01
+    option_gusto = 0x02
+    option_zeppy = 0x03
+    option_darbi = 0x04
+    option_sully = 0x05
+    option_blaze = 0x06
+    default = option_jelze
+
+
+class StartingParty(RandomGroupsChoice):
```

Comment:
This setting allocates a lot of memory for each option permutation, of which there are over a 1000 of. It would be preferred to have this only store the combinations (in sorted order). 

Then if a player puts in an unsorted permutation, sort it during parse so it matches one of the combinations when we do the text lookup.